packages/@lwc/engine-core/src/framework/modules/attr-unless-prop.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import {
+    htmlAttributeToProperty,
+    isNull,
+    isUndefined,
+    StringCharCodeAt,
+    XML_NAMESPACE,
+    XLINK_NAMESPACE,
+} from '@lwc/shared';
+
+import { RendererAPI } from '../renderer';
+import { EmptyObject } from '../utils';
+import { VBaseElement } from '../vnodes';
+
+const ColonCharCode = 58;
+
+export function patchAttrUnlessProp(
+    oldVnode: VBaseElement | null,
+    vnode: VBaseElement,
+    renderer: RendererAPI
+) {
+    const {
+        data: { attrs },
+        elm,
+    } = vnode;
+
+    if (isUndefined(attrs)) {
+        return;
+    }
+
+    const { removeAttribute, setAttribute } = renderer;
+    const oldAttrs = isNull(oldVnode) ? EmptyObject : oldVnode.data.attrs;
+
+    for (const name in attrs) {
+        const cur = attrs[name];
+        const old = oldAttrs[name];
+
+        const propName = htmlAttributeToProperty(name);
+        if (propName in elm!) {
+            (elm as any)[propName] = cur;
```

Comment:
if the property exists, this will always set the prop, independently if the value have changed or not.

Reply:
Ok, so this operation should come after the `old !== cur` check?

Reply:
yes